### PR TITLE
Remove default check for is_set

### LINKED ISSUE #52: A field removed from the rules is still tested
#### Describe the bug
I also to remove a field from the rules, e.g. mounting_position_rmse from the sensor view, but I still get a warning:

```
SensorView.mounting_position_rmse.is_set(None) does not comply in SensorView
```

#### Describe how to reproduce the bug
1. Run the example trace file and the warning will appear.
2. Open rules/osi_sensorview.yml
3. Delete the line "mounting_position_rmse:"
4. Run the example trace file again. The warning will still appear.

#### Describe the expected behavior
I would expect the warning to not appear anymore, because there is no rule defining the mounting_position_rmse anymore. I am not sure, where this warning is coming from. Are the rules cached somewhere?

#### Describe the OS you are using
 - OS: Ubuntu 22.04

### PR DESCRIPTION
**Pull request by ClemensLinnhoff**
#### Reference to a related issue in the repository
#51 
also fixes #52 

#### Add a description
Remove automatically checking for is_set. Now only explicit is_set rules are checked. Also, is_valid does not generate a warning for a rule violation of a subfield. This is somewhat redundant.

#### Mention a member
@jdsika 

#### Check the checklist

- [x] My code and comments follow the [contributors guidelines](https://opensimulationinterface.github.io/osi-documentation/osi/howtocontribute.html) of this project.
- [x] I have performed a self-review of my own code.
- [x] I have made corresponding changes to the [documentation](https://github.com/OpenSimulationInterface/osi-documentation) for osi-validation.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests / travis ci pass locally with my changes.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -78,5 +78,5 @@ jobs:
       run: |
         source .venv/bin/activate
         pip install .
-        osivalidator --data data/20210818T150542Z_sv_312_50_one_moving_object.osi -r rules
-        osivalidator --data data/20210818T150542Z_sv_312_50_one_moving_object.osi -r rules --parallel
+        osivalidator --data data/20240221T141700Z_sv_312_10_one_moving_object.osi -r rules
+        osivalidator --data data/20240221T141700Z_sv_312_10_one_moving_object.osi -r rules --parallel
```
Comment (jdsika) [inline .github/workflows/ci.yml:82]: --parallel can be removed as it is ignored now?

File: README.md [hunk 2]
```diff
@@ -80,5 +80,5 @@ $ source .venv/Scripts/activate
 ## Example command
 
 ```bash
-$ osivalidator --data data/20210818T150542Z_sv_312_50_one_moving_object.osi --rules rules/
+$ osivalidator --data data/20240221T141700Z_sv_312_10_one_moving_object.osi --rules rules/
```
Comment (jdsika) [inline README.md:83]: Was the osi version of the trace too old?

File: doc/usage.adoc [hunk 3]
```diff
@@ -39,23 +39,22 @@ optional arguments:
...
 the path to the trace:
 
 [source,bash]
 ----
-osivalidator --data data/20210818T150542Z_sv_312_50_one_moving_object.osi
+osivalidator --data data/20240221T141700Z_sv_312_10_one_moving_object.osi
 ----
 
-You can also validate the traces in parallel to increase the speed of
+You can also validate the trace in parallel to increase the speed of
```
Comment (jdsika) [inline doc/usage.adoc:52]: I thought that was removed?

File: doc/usage.adoc [hunk 4]
```diff
@@ -232,177 +116,35 @@ Ranges of timestamps    Message
...
-    [0, 498], [500, 998], [1000, 1498],       EnvironmentalConditions.fog.is_set(None) does not comply in SensorView.global_ground_truth.environmental_conditions
-    [1500, 1998], [2000, 2498], [2500, 2716]
-----
-
-The output is generate because of the rules defined in
-`+osi_environment.yml+`:
-
-[source,yaml]
-----
-EnvironmentalConditions:
```
Comment (jdsika) [inline doc/usage.adoc:339]: Don't you think that we have to explain how to work with these rules?

**Comment by jdsika** [comment]
Did you update the osi version?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (jdsika) [reply .github/workflows/ci.yml:82]: We leave it as a test in the CI

File: README.md (hunk 2 above)
Comment (jdsika) [reply README.md:83]: Add OSI version to file name

Comment (ClemensLinnhoff) [reply README.md]: I corrected the OSI version and added the protobuf version"

File: doc/usage.adoc (hunk 3 above)
Comment (ClemensLinnhoff) [reply doc/usage.adoc]: I removed it from the doc.

File: doc/usage.adoc (hunk 4 above)
Comment (ClemensLinnhoff) [reply doc/usage.adoc:339]: The explanation of how to write rules is in "writing-rules.adoc". Here I added some explanation of the rules that are given as errors in the example trace file and removed the rest.

For the future, I would create a new trace file containing more examples of rule violations and then explain them here. But I would do that after the planned major rework of osi-validation.

**Review by jdsika — APPROVED** [review]

**Pull request merged**